src/lib.rs

```diff
@@ -224,24 +224,31 @@ pub trait Interrupts {
     fn get_interrupts(&mut self, clear: bool) -> Result<Self::Irq, Self::Error>;
 }
 
+/// Register contains the address and value of a register.
+///
+/// It is primarily intended as a type constraint for the [Registers] trait.
+pub trait Register: Copy + From<u8> + Into<u8> {
```

Comment:
are we comfortable saying all registers are `u8` width and addressing? it _might_ be possible to make this generic with defaults?

Reply:
(yes we are)

Reply:
As discussed, I used const generics to allow arbitrary size registers. See the test for a usage example.

Reply:
i dug a little more into `modular_bitfields` and it turns out if you annotate `TryFromPrimitive` with `#[repr(u8)]` you get automagic `Into<u8>` and `From<u8>` so it all works out of the box, so it seems like the previous approach is smoother?

Reply:
ah, also this bound should probably be `TryFrom` rather than `From` so we can handle non-exhaustive enum errors (rather than forcing incorrect defaults)